iOSMoviesApp/Controllers/DiscoveryPage/DiscoveryPageViewController.swift

```diff
@@ -0,0 +1,137 @@
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    private var movieListVM : MovieListViewModel!
+    let locationService : LocationService = LocationService()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupLocationService()
+        setupUI()
+    }
+    
+    private func setupUI() {
+        movieListVM = MovieListViewModel(delegate : self)
+        movieListVM.fetchMovies()
+        setupTableView()
+        setupSearchBar()
+    }
+    
+    func setupTableView() {
+        self.tableView.register(UINib(nibName: "MovieCardTableViewCell", bundle: nil), forCellReuseIdentifier: "MovieCardTableViewCell")
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        self.tableView.rowHeight = UITableView.automaticDimension
+    }
+    
+    private func setupSearchBar() {
+        searchBar.delegate = self
+    }
+}
+
+extension DiscoveryPageViewController: UISearchBarDelegate {
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        movieListVM.initializeSearch(for: searchText)
+    }
+
+    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        searchBar.setShowsCancelButton(true, animated: true)
+    }
+
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        searchBar.text = ""
+        searchBar.resignFirstResponder()
+        movieListVM.initializeSearch(for: nil)
+        self.reloadTableView()
+        searchBar.setShowsCancelButton(false, animated: true)
+    }
+}
+
+extension DiscoveryPageViewController: UITableViewDelegate, UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return movieListVM.numberOfMovies()
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cellVM: MovieViewModel = self.movieListVM.getMovieViewModel(at: indexPath)
+        
+        guard let cell: MovieCardTableViewCell = tableView.dequeueReusableCell(withIdentifier: "MovieCardTableViewCell", for : indexPath) as? MovieCardTableViewCell else {
+            self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "DefaultCell")
+            return UITableViewCell()
+        }
+        
+        cell.configureState(with : cellVM, delegate : self, indexPath : indexPath)
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        self.navigateToDetails(for: indexPath)
+    }
+}
+
+extension DiscoveryPageViewController : DiscoveryPageViewControllerDelegate {
+    func navigateToDetails(for indexPath: IndexPath) {
+        let detailsVC: DetailsScreenViewController? = UIStoryboard.init(name : "Main", bundle: Bundle.main).instantiateViewController(withIdentifier: "detailsViewC") as? DetailsScreenViewController
```

Comment:
```suggestion
        guard let detailsVC: DetailsScreenViewController? = UIStoryboard.init(name : "Main", bundle: Bundle.main).instantiateViewController(withIdentifier: "detailsViewC") as? DetailsScreenViewController else {
        return
        }
```
Check if the view controller is `nil` here to avoid unnecessary computations.